tools/c7n_azure/c7n_azure/resources/network_interface.py

```diff
@@ -29,3 +34,52 @@ class resource_type(object):
             'location',
             'resourceGroup'
         )
+
+
+@NetworkInterface.filter_registry.register('effective-route-table')
+class EffectiveRouteTableFilter(ValueFilter):
+    schema = type_schema('effective-route-table', rinherit=ValueFilter.schema)
+
+    def process(self, resources, event=None):
+        futures = []
+        results = []
+        chunk_size = 20
+
+        # Process each resource in a separate thread, returning all that pass filter
+        with self.executor_factory(max_workers=3) as w:
+            for resource_set in chunks(resources, chunk_size):
+                futures.append(w.submit(self.process_resource_set, resource_set))
+
+            for f in as_completed(futures):
+                if f.exception():

```

Comment:
I don't think you need this check.. You handle all exceptions inside `process_resource_set`. If you think this check is valuable, please update log message :)

Reply:
converting to use `ThreadHelper` so I'm refactoring anyway